tests/benchdnn/dnnl_memory.cpp

```diff
@@ -548,6 +558,75 @@ void dnn_mem_t::memset(int value, size_t size) const {
     SAFE_V(FAIL);
 }
 
+dnnl_status_t dnn_mem_t::memset_rng(size_t size) const {
+    bool is_opencl = is_opencl_engine(engine_);
+    bool is_sycl = is_sycl_engine(engine_);
+    auto mem = m_padded_ ? m_padded_ : m_;
+    void *mem_handle;
+    DNN_SAFE_V(dnnl_memory_get_data_handle(mem, &mem_handle));
+    if (is_opencl) {
+#if DNNL_GPU_RUNTIME == DNNL_RUNTIME_OCL
+        stream_t stream(engine_);
+        cl_command_queue ocl_queue;
+        cl_context ocl_ctx;
+        cl_device_id ocl_device;
+        cl_program philox_program;
+        cl_kernel ocl_kernel;
+        cl_int err;
+        DNN_SAFE_V(
+                dnnl_ocl_interop_stream_get_command_queue(stream, &ocl_queue));
+        DNN_SAFE_V(dnnl_ocl_interop_engine_get_context(engine_, &ocl_ctx));
+        DNN_SAFE_V(dnnl_ocl_interop_get_device(engine_, &ocl_device));
+        philox_program = clCreateProgramWithSource(
+                ocl_ctx, 1, &philox_rng_source, nullptr, &err);
+        OCL_TEST(clBuildProgram(
+                philox_program, 1, &ocl_device, nullptr, nullptr, nullptr));
+        ocl_kernel = clCreateKernel(philox_program, "philox_fill_kernel", &err);
+
+        //TODO: generate random seed to avoid repeating values
+        uint64_t seed = 12345;
+        const uint64_t buffSize = (uint64_t)size;
```

Comment:
I'd put an assert that `size` is less than UINT64_MAX to make sure this cast is valid.